packages/zoe/src/contractFacet/zcfSeat.js

```diff
@@ -289,6 +300,91 @@ export const createSeatManager = (
+         * individual `TransferPart` is one of
+         * - A transfer from a `fromSeat` to a `toSeat`. Specify both toAmount
+         *     and fromAmount to change keywords, otherwise only fromAmount is required.
+         * - A taking from a `fromSeat`'s allocation. See the `fromOnly` helper.
+         * - A giving into a `toSeat`'s allocation. See the `toOnly` helper.
+         *
+         * @param {TransferPart[]} transfers
+         */
+        atomicRearrange(transfers) {
+          const newAllocations = makeAllocationMap(transfers);
+
+          // ////// All Seats are active /////////////////////////////////
+          newAllocations.forEach(([seat]) => {
+            assertActive(seat);
+            !seat.hasStagedAllocation() ||
+              Fail`Cannot mix atomicRearrange with seat stagings: ${seat}`;
+            zcfSeatToSeatHandle.has(seat) ||
+              Fail`The seat ${seat} was not recognized`;
+          });
+
+          // ////// Ensure that rights are conserved overall /////////////
+          const flattenAllocations = allocations =>
+            allocations.flatMap(Object.values);
+          const previousAmounts = flattenAllocations(
+            newAllocations.map(([seat]) => seat.getCurrentAllocation()),
+          );
+          const newAmounts = flattenAllocations(
+            newAllocations.map(([_, allocation]) => allocation),
+          );
+          assertRightsConserved(previousAmounts, newAmounts);
+
+          // ////// Ensure that offer safety holds ///////////////////////
+          newAllocations.forEach(([seat, allocation]) => {
+            isOfferSafe(seat.getProposal(), allocation) ||
+              Fail`Offer safety was violated by the proposed allocation: ${allocation}. Proposal was ${seat.getProposal()}`;
+          });
+
+          const seatHandleAllocations = newAllocations.map(
+            ([seat, allocation]) => {
+              const seatHandle = zcfSeatToSeatHandle.get(seat);
+              return { seatHandle, allocation };
+            },
+          );
+          try {
+            // No side effects above. All conditions checked which could have
+            // caused us to reject this reallocation. Notice that the current
+            // allocations are captured in seatHandleAllocations, so there must
+            // be no awaits between that assignment and here.
+            //
+            // COMMIT POINT
+            //
+            // The effects must succeed atomically. The call to
+            // replaceAllocations() will be processed in the order of updates
+            // from zcf to zoe, its effects must occur immediately in zoe on
+            // reception, and must not fail.
+            //
+            // Commit the new allocations (currentAllocation is replaced
+            // for each of the seats) and inform Zoe of the new allocation.
+
+            newAllocations.map(([seat, allocation]) =>
```

Comment:
```suggestion
            newAllocations.forEach(([seat, allocation]) =>
```
or
```suggestion
            for (const [seat, allocation] of newAllocations) {
```
Among the higher-order array operations, `forEach` is the only one that does not return some kind of accumulated result. Thus, it is used only for effects. Since this expression's results are not used, `forEach` is all that's needed. Perhaps this same suggestion applies elsewhere?

Because `forEach` is only for effects, as here, I generally prefer to write it as a for loop. Both are idiomatic enough, so whichever you prefer. I like for loops better because
   * natural to terminate early
   * better emphasizes that is imperative code
   * easier to step through in a debugger.

Reply:
That makes sense. When the usage is specifically "do something simple to every member", I prefer `map` or `forEach`, since they make it clear that there's no intended early exit.

I did convert to `forEach`.

Reply:
@turadg is there a way to add a lint warning about using `.map()` and dropping the result?

Reply:
Good question. I didn't find any specifically but it's an unused expression so maybe https://eslint.org/docs/latest/rules/no-unused-expressions would catch it.

There is a lint rule for preferring the loop over forEach: https://github.com/github/eslint-plugin-github/blob/main/docs/rules/array-foreach.md  I'll see about getting that into our config.